### PR TITLE
Refactor: rework `snapshot_uploaders` module to improve genericity

### PR DESCRIPTION
**Pull request by dlachaume**
## Content

This PR includes a refactor of the `snapshot_uploaders` module and its sub-modules. This is preparatory work to start implementing the sub-builders and uploaders in the issue https://github.com/input-output-hk/mithril/issues/2151.
It involves renaming to remove references to snapshots. The GCP module has been revised to implement directly the `FileUploader` trait.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)

Relates to #2151

### CODE REVIEW

**Review by sfauvel — APPROVED** [review]
LGTM

File: mithril-aggregator/src/file_uploaders/remote_file_uploader.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+use slog::{debug, Logger};
+use std::path::Path;
+
+use mithril_common::logging::LoggerExtensions;
+use mithril_common::StdResult;
+
+use crate::file_uploaders::{FileLocation, FileUploader};
+
+/// RemoteSnapshotUploader is a snapshot uploader working using Google Cloud Platform services
+pub struct RemoteSnapshotUploader {
```
Comment (jpraynaud) [inline mithril-aggregator/src/file_uploaders/remote_file_uploader.rs:11]: You forgot to rename this `RemoteSnapshotUploader`

File: mithril-aggregator/src/file_uploaders/file_uploader.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+use async_trait::async_trait;
+use mithril_common::StdResult;
+use std::path::Path;
+
+pub type FileLocation = String;
```
Comment (jpraynaud) [inline mithril-aggregator/src/file_uploaders/file_uploader.rs:5]: The name is a bit ambiguous, maybe this is better: `RemoteFileLocation`.
Also, an enum would be better here that an alias on a string.

**Review by jpraynaud — COMMENTED** [review]
Overall looks good :+1: 

Can you avoid the cumbersome repetition of `_file_uploaders.rs` in the files of the `file_uploaders` module and rename `file_uploader.rs` to `interface.rs`?

**Review by Alenar — APPROVED** [review]

File: mithril-aggregator/src/file_uploaders/file_uploader.rs (hunk 2 above)
Comment (dlachaume) [reply mithril-aggregator/src/file_uploaders/file_uploader.rs:5]: Suggestion:
`pub struct FileLocationUri(pub String);`

Comment (dlachaume) [reply mithril-aggregator/src/file_uploaders/file_uploader.rs]: Done ✅ , we've replaced it with `pub struct FileUri(pub String);`

File: mithril-aggregator/src/file_uploaders/remote_file_uploader.rs (hunk 1 above)
Comment (dlachaume) [reply mithril-aggregator/src/file_uploaders/remote_file_uploader.rs]: `RemoteSnapshotUploader` was removed to use `GcpUploader` instead.

File: mithril-aggregator/src/file_uploaders/gcp_uploader.rs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        uri.push(filename);
+
+        FileUri(format!("https://{}", uri.join("/")))
+    }
+}
+
+#[async_trait]
+impl FileUploader for GcpUploader {
+    async fn upload(&self, filepath: &Path) -> StdResult<FileUri> {
+        if env::var("GOOGLE_APPLICATION_CREDENTIALS_JSON").is_err() {
```
Comment (jpraynaud) [inline mithril-aggregator/src/file_uploaders/gcp_uploader.rs:47]: Maybe we will have to do this in the dependency builder in a future PR?

File: mithril-aggregator/src/file_uploaders/interface.rs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+use async_trait::async_trait;
+use mithril_common::StdResult;
+use std::path::Path;
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct FileUri(pub String);
```
Comment (jpraynaud) [inline mithril-aggregator/src/file_uploaders/interface.rs:6]: Maybe a comment would be helpful for this type?

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

File: mithril-aggregator/src/file_uploaders/interface.rs (hunk 4 above)
Comment (dlachaume) [reply mithril-aggregator/src/file_uploaders/interface.rs:7]: Done ✅

File: mithril-aggregator/src/file_uploaders/gcp_uploader.rs (hunk 3 above)
Comment (dlachaume) [reply mithril-aggregator/src/file_uploaders/gcp_uploader.rs:47]: Okay, that could be a good idea.

**Pull request merged**